README.md

```diff
@@ -10,21 +10,338 @@ JavaScript XHR
+  console.log(this.responseText)
+  var repoList = "<ul>"
+  for(var i=0;i < this.responseText.length; i++) {
+    repoList += "<li>" + this.responseText[i]["name"] + "</li>"
+  }
+  repoList += "</ul>"
+  document.getElementById("repositories").innerHTML = repoList
+}
+```
+
+Now if we reload and click our link...
+
+![jim hides](http://i.giphy.com/FGTVmzksb2j0k.gif)
+
+Okay not quite what we expected. While it might be fun to have a list of
+a million `undefined` values on a page, we got repositories to print
+out. What happened?
+
+The key lies in the `responseText` property. We can look at it and
+understand that it's JSON, but to our JavaScript interpreter, it's just
+a string of text. And while we *know* that all JSON is just a string of
+text, we have to *tell* JavaScript that it's working with JSON.
+
+This might seem annoying, but honestly, if a computer can't
+innately know that some text is really JSON, it also probably can't become
+sentient. And that's good for all of us.
+
+![thumbs up](http://i.giphy.com/gFwZfXIqD0eNW.gif)
+
+The way we tell the interpreter that we're working with JSON is to parse
+it with [`JSON.parse`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse).
+
+```js
+function showRepositories(event, data) {
+  var repos = JSON.parse(this.responseText)
+  console.log(repos)
+  var repoList = "<ul>"
+  for(var i=0;i < repos.length; i++) {
+    repoList += "<li>" + repos[i]["name"] + "</li>"
+  }
+  repoList += "</ul>"
+  document.getElementById("repositories").innerHTML = repoList
+}
+```
+
+Now we're properly parsing the text into an array of objects that we can
+work with. If we reload and try it again, we should get our list of
+repository names.
+
+Okay, let's take this a step further and set ourselves up to make
+another XHR request based on our data.
+
+We want to be able to list the commits for any given repository. Again,
+we don't want to just re-query the server for each repository as we're
+processing that data, we just want to respond to the user asking for a
+specific repo's commits.
+
+Let's go back into the Github API docs for [commits](https://developer.github.com/v3/repos/commits/) and check it out.
+
+We can see that we can make another `GET` request to `/repos/:owner/:repo/commits` and list the commits. We have the repo name to fill in for the `:repo` parameter based on our repo list. Let's say "repo" a few more times for fun then see what we can do.
+
+We know we'll need an element to click for each repository on our page
+that will request that repository's commits. So we'll need to add a "Get
+Commits" link to our output in `showRepositories`, make a new XHR
+request when that link is clicked, and then show the commits in the
+second column.
+
+We'll start by adding the link to our repository output.
+
+```js
+function showRepositories(event, data) {
+  var repos = JSON.parse(this.responseText)
+  console.log(repos)
+  var repoList = "<ul>"
```

Comment:
one of the next lessons in this unit is about using handlebars templates with this, so these would basically be duplicates of each other.

i think I might be a little confused about just how modular these are supposed to be